tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php

```diff
@@ -1151,4 +1157,75 @@ private function getCurrency(string $reference): Currency
     {
         return new Currency(SharedStorage::getStorage()->get($reference));
     }
+
+    /**
+     * @When I delete cart :cartReference
+     */
+    public function iDeleteCart(string $cartReference)
+    {
+        $cartId = (int) SharedStorage::getStorage()->get($cartReference);
+        try {
+            $this->getCommandBus()->handle(new DeleteCartCommand($cartId));
+        } catch (CartException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then cart :cartReference should be deleted
+     */
+    public function cartShouldBeDeleted(string $cartReference): void
+    {
+        $cartId = (int) SharedStorage::getStorage()->get($cartReference);
+        $repository = new CartRepository();
+
+        try {
+            $repository->get(new CartId($cartId));
+        } catch (CartException $e) {
+            $this->setLastException($e);
+        }
+
+        $this->assertLastErrorIs(CartNotFoundException::class);
+    }
+
+    /**
+     * @Then cart :cartReference should not be deleted because cart is already ordered
+     */
+    public function cartShouldNotBeDeletedBecauseCartIsAlreadyOrdered(string $cartReference)
+    {
+        $this->assertLastErrorIs(CannotDeleteOrderedCartException::class);
+        $this->cartShouldNotBeDeleted($cartReference);
+    }
+
+    /**
+     * @When I bulk delete following carts:
+     */
+    public function iBulkDeleteFollowingCarts(TableNode $table)
+    {
+        $cartIds = [];
+        foreach ($table->getRows() as $row) {
+            $cartIds[] = (int) SharedStorage::getStorage()->get($row[0]);
+        }
+
+        try {
+            $this->getCommandBus()->handle(new BulkDeleteCartCommand($cartIds));
+        } catch (CartException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Given cart :cartReference should not be deleted
```

Comment:
```suggestion
     * @Given cart :cartReference should exist
```
this way is applicable to more scenarios without adding additional definitions :man_shrugging: